zcash_client_backend/src/tor/http.rs

```diff
@@ -57,15 +58,47 @@ impl Client {
         f: impl FnOnce(Incoming) -> F,
         retry_limit: u8,
     ) -> Result<Response<T>, Error> {
+        self.http_request(
+            url,
+            |builder| h(builder).method("GET"),
+            Empty::<Bytes>::new(),
+            f,
+            retry_limit,
+        )
+        .await
+    }
+
+    /// Makes an HTTP request over Tor.
+    ///
+    /// On error, retries will be attempted as follows:
+    /// - A successful request that resulted in a client error (HTTP 400-499) will cause a
+    ///   retry with an isolated client.
```

Comment:
We should probably choose a subset of this range that will result in retries; for example, 403 and 404 almost certainly should not be retried.

Reply:
IIRC 403 is the most common error code I see when a Tor exit node is blocked, and is precisely when I want this to trigger.

Reply:
Okay; maybe 404 is worth respecting though?

Reply:
I think if we're going to filter it to less than "all 4XX status codes", then we probably should make the set of status codes configurable by the user, which makes the API more complicated. The current behaviour was fine when we just exposed the limited "fetch exchange rate" API, as we knew the kinds of status codes to expect.

I'll try and come up with an alternative API that isn't horrible.